### PR TITLE
Fine grained admin permissions feature V2

### LINKED ISSUE #34563: [FGAP] Add feature V2
Current feature is in `PREVIEW` state https://github.com/keycloak/keycloak/blob/a7af380f71076b3a5b9372bee955541bfd1fd11a/common/src/main/java/org/keycloak/common/Profile.java#L56

To avoid disrupting existing deployments that rely on this feature, we should preserve its current implementation (perhaps make it `DEPRECATED` in future). 

Instead, a new version of the feature should be created. Additionally, it might be beneficial to protect existing FGAP endpoints with a feature flag to enhance control over feature availability.

### LINKED ISSUE #34635: Feature in higher version takes precedence even if it has lower type order
The feature type should have had higher priority than its version, so that if the feature is e.g. `EXPERIMENTAL` it should not be picked if there is the feature in lower version, but higher type (e.g. `PREVIEW`)

### PR DESCRIPTION
**Pull request by vramik**
Closes #34563
Closes #34635

### CODE REVIEW

File: common/src/main/java/org/keycloak/common/Profile.java [hunk 1]
```diff
@@ -53,7 +53,9 @@ public enum Feature {
 
         ACCOUNT_V3("Account Console version 3", Type.DEFAULT, 3, Feature.ACCOUNT_API),
 
-        ADMIN_FINE_GRAINED_AUTHZ("Fine-Grained Admin Permissions", Type.PREVIEW),
+        ADMIN_FINE_GRAINED_AUTHZ("Fine-Grained Admin Permissions", Type.PREVIEW, 1),
```
Comment (pedroigor) [inline common/src/main/java/org/keycloak/common/Profile.java:56]: We should have this feature starting as `EXPERIMENTAL`.

Comment (pedroigor) [reply common/src/main/java/org/keycloak/common/Profile.java:56]: The `v1` version should be the default when starting the server using `fine-grained-admin, doesn't it?

Comment (vramik) [reply common/src/main/java/org/keycloak/common/Profile.java:56]: > The `v1` version should be the default when starting the server using `fine-grained-admin, doesn't it?

The preview feature should take a precedence over experimental, if unversioned key is used: https://github.com/keycloak/keycloak/blob/ea5e623b367a27511d881249695077d860559f4c/common/src/main/java/org/keycloak/common/Profile.java#L325-L336

But it seems the current behavior is not correct, I'll try to include the fix as part of this PR

Comment (vramik) [reply common/src/main/java/org/keycloak/common/Profile.java:56]: with the second commit, it should be fixed

**Review by sguilhen — APPROVED** [review]

**Review by ahus1 — APPROVED** [review]
Approving based on previous reviews.

**Pull request merged**